pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -400,6 +403,50 @@
     '';
   };
 
+  codesnap-nvim =
+    let
+      version = "1.3.1";
+      src = fetchFromGitHub {
+        owner = "mistricky";
+        repo = "codesnap.nvim";
+        rev = "refs/tags/v${version}";
+        hash = "sha256-nS/bAWsBQ1L4M9437Yp6FdmHoogzalKlLIAXnRZyMp0=";
+      };
+      codesnap-lib = rustPlatform.buildRustPackage {
+        pname = "codesnap-lib";
+        inherit version src;
+
+        sourceRoot = "${src.name}/generator";
+
+        cargoHash = "sha256-FTQl5WIGEf+RQKYJ4BbIE3cCeN+NYUp7VXIrpxB05tU=";
+
+        nativeBuildInputs = [
+          pkg-config
+          rustPlatform.bindgenHook
+        ];
+
+        buildInputs = [
+          libuv.dev
+        ] ++ lib.optionals stdenv.isDarwin [
+          darwin.apple_sdk.frameworks.AppKit
+        ];
+      };
+    in
+    buildVimPlugin {
+      pname = "codesnap.nvim";
+      inherit version src;
+
+      # https://github.com/mistricky/codesnap.nvim/blob/main/scripts/build_generator.sh
+      postInstall = let
+        extension = if stdenv.isDarwin then "dylib" else "so";
```

Comment:
you can use sthg like `stdenv.hostPlatform.extensions.sharedLibrary` instead. in this case I guess it should be `targetPlatform` ? I dont know ^^' but I think hostPlatform is for the builder.

Reply:
According to the [docs](https://github.com/NixOS/nixpkgs/blob/138408df0468e52bf4abefc20a566f405724565e/doc/stdenv/cross-compilation.chapter.md#platform-parameters-ssec-cross-platform-parameters):

> The "host platform" is the platform on which a package will be run.

So I think you're right, there.

Also, doesn't the library need to be copied to `lua/generator.${extension}` instead of `lua/generator.so` ?

Reply:
This is what I thought, but looking at the [script](https://github.com/mistricky/codesnap.nvim/blob/main/scripts/build_generator.sh) suggests that the destination should always be `generator.so`...
But now that I have a look at it again, it feels like a mistake...
Could you please help us with this @mistricky ?

Reply:
Hi @GaetanLepage, sorry to reply late, wow thank u for packing codesnap to nixpkgs. 

The `generator.so` is build from rust source, and called by Lua, both on Mac OS and Linux, the suffix should be `.so`, so the [script](https://github.com/mistricky/codesnap.nvim/blob/main/scripts/build_generator.sh) just rename the `generator.[dylib | so]` to `generator.so`, it's not a mistake. 

But there is another situation, if the user on Windows, the suffix should be `.dll`, but CodeSnap does not support Windows platform yet, so the filename of `generator` should always be`lua/generator.so` for now.

Reply:
Ok, very good to know, thanks for coming in and helping !
I went and trust the script, so we have nothing to change.